core/src/main/java/io/grpc/ExponentialBackoffPolicy.java

```diff
@@ -0,0 +1,149 @@
+        .initialBackoffMilis(TimeUnit.SECONDS.toMillis(1))
+        .maxBackoffMillis(TimeUnit.MINUTES.toMillis(2))
+        .multiplier(1.6)
+        .jitter(.2)
+        .build();
+    }
+  }
+
+  private final Random random;
+  private final long initialBackoffMillis;
+  private final long maxBackoffMillis;
+  private final double multiplier;
+  private final double jitter;
+
+  private long nextBackoffMillis;
+
+  private ExponentialBackoffPolicy(Random random, long initialBackoffMilis, long maxBackoffMillis,
+                                   double multiplier, double jitter) {
+    this.random = random;
+    this.initialBackoffMillis = initialBackoffMilis;
+    this.maxBackoffMillis = maxBackoffMillis;
+    this.multiplier = multiplier;
+    this.jitter = jitter;
+
+    this.nextBackoffMillis =  initialBackoffMillis;
+  }
+
+  @Override
+  public long nextBackoffMillis() {
+    long currentBackoffMillis = nextBackoffMillis;
+    nextBackoffMillis = Math.min((long) (currentBackoffMillis * multiplier), maxBackoffMillis);
+    return currentBackoffMillis
+        + uniformRandom(-jitter * currentBackoffMillis, jitter * currentBackoffMillis);
+  }
+
+  private long uniformRandom(double low, double high) {
+    checkArgument(high >= low);
+    double mag = high - low;
+    return (long) (random.nextDouble() * mag + low);
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static final class Builder {
+    private Random random;
+    private long initialBackoffMilis;
+    private long maxBackoffMillis;
+    private double multiplier;
+    private double jitter;
+
+    private Builder() {
+    }
+
+    public Builder random(Random random) {
+      checkNotNull(random);
+      return this;
+    }
+
+    public Builder initialBackoffMilis(long initialBackoffMilis) {
+      checkState(initialBackoffMilis > 0, "initialBackoffMilis must be greater than zero");
+      this.initialBackoffMilis = initialBackoffMilis;
+      return this;
+    }
+
+    public Builder maxBackoffMillis(long maxBackoffMillis) {
+      checkState(maxBackoffMillis > 0, "maxBackoffMillis must be greater than zero");
+      this.maxBackoffMillis = maxBackoffMillis;
+      return this;
+    }
+
+    public Builder multiplier(double multiplier) {
+      checkState(multiplier > 0, "multiplier must be greater than zero");
+      this.multiplier = multiplier;
+      return this;
+    }
+
+    public Builder jitter(double jitter) {
+      checkState(jitter >= 0, "jitter must be greater or equal zero");
+      this.jitter = jitter;
+      return this;
+    }
+
+    public ExponentialBackoffPolicy build() {
+      checkState(initialBackoffMilis < maxBackoffMillis, "initialBackoffMilis can't be greater than"
+              + "than maxBackoffMillis");
+      random = firstNonNull(this.random, new Random());
+
+      return new ExponentialBackoffPolicy(random, initialBackoffMilis, maxBackoffMillis, multiplier,
```

Comment:
You can avoid the long number of parameters by passing "this" into the Constructor, and having it extract the values.  It also saves you from having the "this.foo = foo" idiom all over the constructor.